### PR TITLE
Make shutil rename in python_only_dev

### PR DESCRIPTION
**Pull request by shcheglovnd**
- Use shutil to rename in python_only_dev to support rename between different devices in linux

### CODE REVIEW

**Comment by russellb** [comment]
This overlaps with #9583

**Comment by shcheglovnd** [comment]
> This overlaps with #9583

Thank you. I will remain only rename fix here then and add in #9583 a comment.

File: python_only_dev.py [hunk 1]
```diff
@@ -69,7 +69,8 @@
     current_vllm_path = os.path.join(cwd, "vllm")
 
     print(f"Renaming {pre_built_vllm_path} to {tmp_path} for backup")
-    os.rename(pre_built_vllm_path, tmp_path)
```
Comment (youkaichao) [inline python_only_dev.py:72]: is it mentioned anywhere that this does not support rename between different devices in linux?

Comment (shcheglovnd) [reply python_only_dev.py:72]: https://docs.python.org/3/library/os.html#os.rename

> On Windows, if dst exists a [FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) is always raised. The operation may fail if src and dst are on different filesystems. Use [shutil.move()](https://docs.python.org/3/library/shutil.html#shutil.move) to support moves to a different filesystem.

**Review by youkaichao — APPROVED** [review]
thanks for the fix!

**Comment by pichiu** [comment]
Hi, noticed that there's another `os.rename` in the recovering backup section at [L92](https://github.com/vllm-project/vllm/pull/10233/files#diff-f77130d93453aef639a74d0217ab7a1d9ea0667da14fde0d3244bc9cc8e640e7R92). Will this one be updated as well? Thanks

**Comment by youkaichao** [comment]
@pichiu contribution is welcome!

**Pull request merged**